### ISSUE TITLE
[Bug]: AutoFill is not offered during MacOS logins

### ISSUE THREAD

**Issue by cassiocassio** [mattermost/desktop#2914]
### Checks before filing an issue

- [X] This issue doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/mattermost-server/issues).
- [X] I have checked the [issue tracker](https://github.com/mattermost/desktop/issues) and have not found an issue that matches the one I'm filing.
- [X] This issue is not a troubleshooting question. Troubleshooting questions go here: https://forum.mattermost.com/c/trouble-shoot/16.
- [X] This issue is not a feature request. You can request features and make product suggestions here: https://mattermost.com/suggestions/.
- [X] This issue reproduces on the most recent [stable version](https://github.com/mattermost/desktop/releases/latest), or the most recent [prerelease version](https://github.com/mattermost/desktop/releases) of the Mattermost Desktop App.
- [X] I have read the [contribution guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

### Mattermost Desktop Version

Version 5.5.1 (63830002741)

### Operating System

MacOS 14.0 (23A344

### Mattermost Server Version

Mattermost Version: 8.1.3 Database Schema Version: 113

### Steps to reproduce

Log out and log into Mattermost server from MacOS Desktop Client

### Expected behavior

Using the web client both Chrome and Safari browsers on MacOS autofill passwords via the Chrome Password manager or the Apple system Keychain:

Safari:
<img width="562" alt="safari" src="https://github.com/mattermost/desktop/assets/4593556/d07e2556-b8c9-4755-b795-183471eae0b7">

Chrome:
<img width="600" alt="chrome" src="https://github.com/mattermost/desktop/assets/4593556/fdd36c65-f704-4663-9f3b-0b26ae6cb401">

### Observed behavior

<img width="599" alt="mm" src="https://github.com/mattermost/desktop/assets/4593556/ce4cb617-82f5-4a0d-8762-a6e245d90bfa">

One the MacOS Desktop App Username and Password AutoFill is not offered during the login process.

Keychain doesn't offer to i. remember credentials  nor  ii. autofill 

It can't be invoked manually from the Edit > AutoFill > Passwords menu either.

 

### Log Output

```shell
not sure this is relevant:

Date/Time:        2023-11-13 22:32:31.250 +0000
End time:         2023-11-14 17:09:29.743 +0000
OS Version:       macOS 14.0 (Build 23A344)
Architecture:     arm64e
Report Version:   44
Incident Identifier: ABCB16C4-F33D-4890-8F1C-7C2B4C31CBEB
Share With Devs:  Yes

Data Source:      Microstackshots
Shared Cache:     7906534C-5D2B-3BA8-B9C6-93A5CF1B7B50 slid base address 0x183158000, slide 0x3158000

Command:          Mattermost
Path:             /Applications/Mattermost.app/Contents/MacOS/Mattermost
Identifier:       Mattermost.Desktop
Version:          5.5.1 (63830002741)
Team ID:          UQ8HT4Q2XM
App Item ID:      1614666244
App External ID:  860358278
Is First Party:   No
Resource Coalition ID: 4274
Architecture:     arm64
PID:              4295

Event:            disk writes
Action taken:     none
Writes:           2147.49 MB of file backed memory dirtied over 67018 seconds (32.04 KB per second average), exceeding limit of 24.86 KB per second over 86400 seconds
Writes limit:     2147.48 MB
Limit duration:   86400s
Writes caused:    2147.49 MB
Writes duration:  67018s
Duration:         67018.49s
Duration Sampled: 24087.31s
Steps:            6 (10.49 MB/step)

Hardware model:   Mac14,6
Active cpus:      12
HW page size:     16384
VM page size:     16384

OS Cryptex File Extents: 168

Heaviest stack for the target process:
  6  thread_start + 8 (libsystem_pthread.dylib + 7740) [0x183589e3c]
  6  _pthread_start + 136 (libsystem_pthread.dylib + 28724) [0x18358f034]
  6  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18248376 (Electron Framework + 50706944) [0x115db7a00]
  6  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18139432 (Electron Framework + 50598000) [0x115d9d070]
  6  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18139728 (Electron Framework + 50598296) [0x115d9d198]
  6  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18140436 (Electron Framework + 50599004) [0x115d9d45c]
  4  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18079484 (Electron Framework + 50538052) [0x115d8e644]
  4  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18083044 (Electron Framework + 50541612) [0x115d8f42c]
  4  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18081776 (Electron Framework + 50540344) [0x115d8ef38]
  4  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 17947112 (Electron Framework + 50405680) [0x115d6e130]
  2  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33633328 (Electron Framework + 66091896) [0x116c63b78]
  1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33634164 (Electron Framework + 66092732) [0x116c63ebc]
  1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33627092 (Electron Framework + 66085660) [0x116c6231c]
  1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33603084 (Electron Framework + 66061652) [0x116c5c554]
  1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33742944 (Electron Framework + 66201512) [0x116c7e7a8]
  1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18282740 (Electron Framework + 50741308) [0x115dc003c]
  1  __unlink + 8 (libsystem_kernel.dylib + 22604) [0x18355284c]


Powerstats for:   Mattermost [4295]
UUID:             4C4C44CB-5555-3144-A155-857479BAA7D7
Path:             /Applications/Mattermost.app/Contents/MacOS/Mattermost
Identifier:       Mattermost.Desktop
Version:          5.5.1 (63830002741)
Team ID:          UQ8HT4Q2XM
App Item ID:      1614666244
App External ID:  860358278
Is First Party:   No
Resource Coalition ID: 4274
Architecture:     arm64
Footprint:        102.38 MB -> 116.50 MB (+14.12 MB)
Pageins:          8 pages
Start time:       2023-11-14 10:25:02.149 +0000
End time:         2023-11-14 17:06:29.462 +0000
Num samples:      6 (100%)
Primary state:    4 samples Frontmost App, Non-Suppressed, Kernel mode, Effective Thread QoS User Initiated, Requested Thread QoS User Initiated, Override Thread QoS Unspecified
User Activity:    0 samples Idle, 6 samples Active
Power Source:     3 samples on Battery, 3 samples on AC
  6  thread_start + 8 (libsystem_pthread.dylib + 7740) [0x183589e3c]
    6  _pthread_start + 136 (libsystem_pthread.dylib + 28724) [0x18358f034]
      6  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18248376 (Electron Framework + 50706944) [0x115db7a00]
        6  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18139432 (Electron Framework + 50598000) [0x115d9d070]
          6  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18139728 (Electron Framework + 50598296) [0x115d9d198]
            6  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18140436 (Electron Framework + 50599004) [0x115d9d45c]
              4  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18079484 (Electron Framework + 50538052) [0x115d8e644]
                4  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18083044 (Electron Framework + 50541612) [0x115d8f42c]
                  4  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18081776 (Electron Framework + 50540344) [0x115d8ef38]
                    4  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 17947112 (Electron Framework + 50405680) [0x115d6e130]
                      2  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33633328 (Electron Framework + 66091896) [0x116c63b78]
                        1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33634164 (Electron Framework + 66092732) [0x116c63ebc]
                          1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33627092 (Electron Framework + 66085660) [0x116c6231c]
                            1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33603084 (Electron Framework + 66061652) [0x116c5c554]
                              1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33742944 (Electron Framework + 66201512) [0x116c7e7a8]
                                1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18282740 (Electron Framework + 50741308) [0x115dc003c]
                                  1  __unlink + 8 (libsystem_kernel.dylib + 22604) [0x18355284c]
                        1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33631616 (Electron Framework + 66090184) [0x116c634c8]
                          1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33630460 (Electron Framework + 66089028) [0x116c63044]
                            1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33731592 (Electron Framework + 66190160) [0x116c7bb50]
                              1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33614812 (Electron Framework + 66073380) [0x116c5f324]
                                1  write + 8 (libsystem_kernel.dylib + 21460) [0x1835523d4]
                                  1  <Non-Frontmost App>
                      1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 5645276 (Electron Framework + 38103844) [0x1151b2b24]
                        1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 5635080 (Electron Framework + 38093648) [0x1151b0350]
                          1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 5455932 (Electron Framework + 37914500) [0x115184784]
                            1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 5563220 (Electron Framework + 38021788) [0x11519ea9c]
                              1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 5727272 (Electron Framework + 38185840) [0x1151c6b70]
                                1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 5723628 (Electron Framework + 38182196) [0x1151c5d34]
                                  1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 5438912 (Electron Framework + 37897480) [0x115180508]
                                    1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 5415560 (Electron Framework + 37874128) [0x11517a9d0]
                                      1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 5414544 (Electron Framework + 37873112) [0x11517a5d8]
                                        1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33742944 (Electron Framework + 66201512) [0x116c7e7a8]
                                          1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18282740 (Electron Framework + 50741308) [0x115dc003c]
                                            1  __unlink + 8 (libsystem_kernel.dylib + 22604) [0x18355284c]
                      1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 3793988 (Electron Framework + 36252556) [0x114feeb8c]
                        1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 10420968 (Electron Framework + 42879536) [0x115640a30]
                          1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 10421104 (Electron Framework + 42879672) [0x115640ab8]
                            1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 10520072 (Electron Framework + 42978640) [0x115658d50]
                              1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 10428224 (Electron Framework + 42886792) [0x115642688]
                                1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33618524 (Electron Framework + 66077092) [0x116c601a4]
                                  1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33640484 (Electron Framework + 66099052) [0x116c6576c]
                                    1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33659516 (Electron Framework + 66118084) [0x116c6a1c4]
                                      1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33659964 (Electron Framework + 66118532) [0x116c6a384]
                                        1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 33616772 (Electron Framework + 66075340) [0x116c5facc]
                                          1  write + 8 (libsystem_kernel.dylib + 21460) [0x1835523d4]
                                            1  <Non-Frontmost App>
              2  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18079612 (Electron Framework + 50538180) [0x115d8e6c4]
                2  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18082428 (Electron Framework + 50540996) [0x115d8f1c4]
                  2  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 17947112 (Electron Framework + 50405680) [0x115d6e130]
                    2  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18157780 (Electron Framework + 50616348) [0x115da181c]
                      2  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18157280 (Electron Framework + 50615848) [0x115da1628]
                        2  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 37308976 (Electron Framework + 69767544) [0x116fe5178]
                          1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 37312028 (Electron Framework + 69770596) [0x116fe5d64]
                            1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 18244564 (Electron Framework + 50703132) [0x115db6b1c]
                              1  __fcntl + 8 (libsystem_kernel.dylib + 9244) [0x18354f41c]
                          1  v8::internal::SetupIsolateDelegate::SetupBuiltins(v8::internal::Isolate*, bool) + 37308736 (Electron Framework + 69767304) [0x116fe5088]
                            1  write + 8 (libsystem_kernel.dylib + 21460) [0x1835523d4]

  Binary Images:
           0x100e2c000 -                ???  Mattermost.Desktop 5.5.1 (63830002741) <4C4C44CB-5555-3144-A155-857479BAA7D7>  /Applications/Mattermost.app/Contents/MacOS/Mattermost
           0x112d5c000 -        0x11afcffff  com.github.Electron.framework (26.2.1) <4C4C44A5-5555-3144-A1C5-265B058B90DE>  /Applications/Mattermost.app/Contents/Frameworks/Electron Framework.framework/Versions/A/Electron Framework
           0x18354d000 -        0x183587fef  libsystem_kernel.dylib (10002.1.13)    <A7D3C07D-0A1E-3C4C-8FBA-66905E16BF99>  /usr/lib/system/libsystem_kernel.dylib
           0x183588000 -        0x183594ff3  libsystem_pthread.dylib (519)          <E4DEBB6E-421D-33D0-9E17-77AE0E0FE4DC>  /usr/lib/system/libsystem_pthread.dylib

System Profile:
Network Service: Wi-Fi, AirPort, en0
Thunderbolt Bus: MacBook Pro, Apple Inc.
Thunderbolt Bus: MacBook Pro, Apple Inc.
Thunderbolt Bus: MacBook Pro, Apple Inc.
Boot Volume File System Type: apfs
Memory Module: LPDDR5, Micron
USB Device: USB31Bus
USB Device: iPhone
USB Device: USB31Bus
USB Device: MagSafe Charging Case
USB Device: USB31Bus
Display: Color LCD, 3456 x 2234 Retina, Main, MirrorOff, Online
Model: Mac14,6, BootROM 10151.1.1, proc 12:8:4 processors, 32 GB, SMC 
Graphics: Apple M2 Max, Apple M2 Max, Built-In
AirPort: spairport_wireless_card_type_wifi (0x14E4, 0x4388), wl0: Aug  8 2023 04:53:09 version 23.10.745.4.41.51.118 FWID 01-8492d501
AirPort: 
Bluetooth: Version (null), 0 services, 0 devices, 0 incoming serial ports
```

### Additional Information

I'm not sure if this is a Mac issue / an Electron issue / a by-design choice

**Comment by devinbinnie** [comment]
@cassiocassio Can you try the latest nightly build? Seems to be working for me on there :)
https://community.mattermost.com/core/channels/desktop-builds

**Comment by devinbinnie** [comment]
Closing as inactive/cannot reproduce. Should be fixed in v5.6
